meta-mender-toradex-nxp/classes/mender_tezimg.bbclass

```diff
@@ -94,53 +99,51 @@ python mender_toradex_json() {
+        mtddevs = mender_rootfs_rawnand(d)
+        if mtddevs == None:
+            return # warning message already generated
+        data["mtddevs"] = mtddevs
 
     deploy_dir = d.getVar("DEPLOY_DIR_IMAGE", True)
     image_name = d.getVar("IMAGE_BASENAME", True) + "-" + d.getVar("MACHINE")
 
-    # Copy image file to "toradex" subdirectory (in image deploy directory)
-    dst = os.path.join(deploy_dir, "toradex-easyinstaller")
-    if not os.path.exists(dst):
-        os.makedirs(dst)
-    src_match = re.search("%s-%s-[0-9]{14}\.ubimg" % \
-                         (d.getVar("IMAGE_BASENAME", True), \
-                          d.getVar("MACHINE", True)), \
-                         " ".join(os.listdir(d.getVar('IMGDEPLOYDIR'))))
-    if src_match == None:
-         bb.warn("Could not copy ubimg to toradex-easyinstaller subdir; file not found.")
-    else:
-        src = os.path.join(d.getVar('IMGDEPLOYDIR'), src_match.group(0))
-        shutil.copy(src, os.path.join(dst, image_name) + ".ubimg")
-
-    # Copy uboot environment files
-    _dst_ = os.path.join(dst, "uboot.env")
-    src = os.path.join(deploy_dir, "uboot.env")
-    shutil.copy(src, _dst_)
-
-    _dst_ = os.path.join(dst, "u-boot-nand.imx")
-    src = os.path.join(deploy_dir, "u-boot-nand.imx")
-    shutil.copy(src, _dst_)
-
-    _dst_ = os.path.join(dst, "uEnv.txt")
-    src = os.path.join(deploy_dir, "uEnv.txt")
-    shutil.copy(src, _dst_)
-    
     # Write generated image.json file
-    with open(os.path.join(dst, "image.json"), "w") as outfile:
+    with open(os.path.join(deploy_dir, "image.json"), "w") as outfile:
         json.dump(data, outfile, indent=4)
     bb.note("Toradex Easy Installer metadata file image.json written.")
-    bb.note("Easy installer deployment-ready directory \"%s\" created." % dst)
 }
 
-# Add JSON-generating function after image has been created
-do_image_complete[postfuncs] += "mender_toradex_json"
+IMAGE_CMD_mender_tezimg() {
+    case "${IMAGE_FSTYPES}" in
+    *"ubimg"*)
+        tar --transform="s,.*/,${IMAGE_BASENAME}-${MACHINE}-Tezimg/," \
+            -chf ${IMGDEPLOYDIR}/${MACHINE}-${IMAGE_BASENAME}-Tezi-${DATETIME}.tar \
+                    ${DEPLOY_DIR_IMAGE}/image.json \
+                    ${DEPLOY_DIR_IMAGE}/mender_toradex_linux.png \
+                    ${DEPLOY_DIR_IMAGE}/marketing_mender_toradex.tar \
+                    ${DEPLOY_DIR_IMAGE}/prepare.sh \
+                    ${DEPLOY_DIR_IMAGE}/wrapup.sh \
+                    ${DEPLOY_DIR_IMAGE}/${UBOOT_BINARY} \
+                    ${DEPLOY_DIR_IMAGE}/uboot.env \
+                    ${DEPLOY_DIR_IMAGE}/uEnv.txt \
+                    ${IMGDEPLOYDIR}/${IMAGE_BASENAME}-${MACHINE}.ubimg
+        bbnote "Mender Tezi metadata successfully written and packed."
+        ;;
+    *)
+        bbwarn "\"ubifs\" not in IMAGE_FSTYPES. Skipping tezimg."
+    esac
+}
+
+IMAGE_FSTYPES += "mender_tezimg"
+IMAGE_FSTYPES_remove = "teziimg"
```

Comment:
Since the original is called `teziimg`, maybe we should stick with two i's in our image too?

Also, do the images conflict? I would not automatically remove the `teziimg`, but rather leave this to the user.